lua/nvim-treesitter/shell_command_selectors.lua

```diff
@@ -96,8 +96,14 @@ function M.select_compiler_args(repo, compiler)
       repo.files,
       "-shared",
       "-Os",
-      "-lstdc++",
     }
+    if
+      #vim.tbl_filter(function(file)
+        return file:match "%.cc$" or file:match "%cpp$" or file:match "%.cxx$"
+      end, repo.files) > 0
+    then
+      table.insert(args, "-lstdc++")
+    end
```

Comment:
I didn't apply the same change for zig, as I assumed that to be much rarer.

Reply:
not on Windows, it seems...

Reply:
For MSVC there is no distinction between C and C++

Reply:
I was specifically talking about  `zig`, which I've seen used on Windows. (Hilarity ensued.)

But I don't think we should support that, really: `zig c++` is _not_ ready for productive use yet.

Reply:
No idea how this works, this is why I didn't change that. If anybody can tell me how I should change that zig line, I'm happy. We would probably need to test zig when we actually want to support it